cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/statements/expressions/BinaryOperator.kt

```diff
@@ -117,20 +119,39 @@ class BinaryOperator : Expression(), HasType.TypeListener, Assignment, HasBase,
         }
         val previous = type
         if (operatorCode == "=") {
+            if (
+                src == rhs &&
```

Comment:
This is a quick fix for #1071 but only holds for numbers. For other types (e.g. classes), we probably have to check what we want to propagate and is somewhat related to #204.